source/CIS/cis-4.rst

```diff
@@ -0,0 +1,750 @@
+
+All of the fields in the JSON file are optional, and this specification reserves a number of field names, shown in the table below.
+
+.. list-table:: Issuer metadata JSON Object
+  :header-rows: 1
+
+  * - Property
+    - JSON value type [JSON-Schema]
+    - Description
+  * - ``name``
+    - string
+    - The name to display for the issuer.
+  * - ``icon``
+    - URL JSON object
+    - An image URL for displaying the issuer.
+  * - ``description`` (optional)
+    - string
+    - A description for the issuer.
+  * - ``url`` (optional)
+    - string (:rfc:`3986`) [``uri-reference``]
+    - A URL of the issuer's website.
+
+Optionally a SHA256 hash of the JSON file can be logged with the :ref:`CIS-4-events-IssuerMetadata` event for checking integrity.
+Since the metadata JSON file could contain URLs, a SHA256 hash can optionally be associated with the URL.
+To associate a hash with a URL the JSON value is an object:
+
+.. list-table:: URL JSON Object
+  :header-rows: 1
+
+  * - Property
+    - JSON value type [JSON-Schema]
+    - Description
+  * - ``url``
+    - string (:rfc:`3986`) [``uri-reference``]
+    - A URL.
+  * - ``hash`` (optional)
+    - string
+    - A SHA256 hash of the URL content encoded as a hex string.
+
+Example issuer metadata
+^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: json
+
+  {
+    "name": "Concordium",
+    "icon" : {
+      "url":  "https://concordium.com/wp-content/uploads/2022/07/Concordium-1.png",
+      "hash": "1c74f7eb1b3343a5834e60e9a8fce277f2c7553112accd42e63fae7a09e0caf8"
+      }
+    "description": "A public-layer 1, science-backed blockchain",
+    "url": "https://concordium.com"
+  }
+
+Credential metadata JSON
+------------------------
+
+The credential metadata is stored off-chain and MUST be a JSON (:rfc:`8259`) file.
+
+.. list-table:: Credential metadata JSON Object
+  :header-rows: 1
+
+  * - Property
+    - JSON value type [JSON-Schema]
+    - Description
+  * - ``title``
+    - string
+    - The name to display for the credential.
+  * - ``logo``
+    - URL JSON object
+    - An image URL for displaying the credential.
+  * - ``background_color``
+    - string
+    - A hex code of the background color for displaying the credential.
+  * - ``image`` (optional)
+    - URL JSON object
+    - A background image URL for displaying the credential.
+
+Where URL JSON object the same as in :ref:`CIS-4-events-IssuerMetadata`.
+
+Optionally a SHA256 hash of the JSON file can be logged with the :ref:`CIS-4-events-CredentialMetadataEvent` event for checking integrity.
+
+
+Example credential metadata
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: json
+
+ {
+    "title": "Concordium Employment",
+    "logo" : {
+      "url":  "https://concordium.com/wp-content/uploads/2022/07/Concordium-1.png",
+      "hash": "1c74f7eb1b3343a5834e60e9a8fce277f2c7553112accd42e63fae7a09e0caf8"
+      }
+    "background_color": "#000000",
+    "image": {
+      "url": "https://concordium.com/employment/vc-background.png",
```

Comment:
There is a hash for the `logo` but not for the `image`. Should it be the same?

Reply:
`hash` is optional, so I wanted to give an example with and without `hash`. I can add a comment explaining this.